mantaray/received.py

```diff
@@ -590,6 +594,9 @@ def _handle_received_message(
     elif msg.command == RPL_TOPIC:
         _handle_numeric_rpl_topic(server_view, msg.args)
 
+    elif msg.command == RPL_WHOISUSER:
+        if msg.args[0] == server_view.core.nick:
+            server_view.core._nickmask = f"!{msg.args[2]}@{msg.args[3]}"
```

Comment:
Naming an attribute `_nickmask` convention is supposed to mean that it is accessed only inside a single class, not outside it. Can you add a `set_nickmask()` method to the core that works like this?

```suggestion
            server_view.core.set_nickmask(user=msg.args[2], host=msg.args[3])
```